libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt

```diff
@@ -113,12 +128,52 @@ class MapboxVoiceInstructionsPlayer @JvmOverloads constructor(
         announcement: SpeechAnnouncement,
         consumer: MapboxNavigationConsumer<SpeechAnnouncement>
     ) {
-        playCallbackQueue.add(PlayCallback(announcement, consumer))
+        play(announcement, consumer, 500)
+    }
+
+    /**
+     * Given [SpeechAnnouncement] the method will play the voice instruction.
+     * If a voice instruction is already playing or other announcement are already queued,
+     * the given voice instruction will be queued to play after.
+     * @param announcement object including the announcement text
+     * and optionally a synthesized speech mp3.
+     * @param consumer represents that the speech player is done playing
+     * @param expirationMillis if the specified timeout passes between the moment the instruction
+     *  was removed from the queue to be played and the moment the playing actually starts,
+     *  it will be cancelled. As an alternative, you may manually invoke the [cancel] method
+     *  as soon as playing is no longer required. If null is provided, it will never be cancelled.
+     */
+    fun play(
+        announcement: SpeechAnnouncement,
+        consumer: MapboxNavigationConsumer<SpeechAnnouncement>,
+        expirationMillis: Long?
+    ) {
```

Comment:
We could do that but:
a) it would require users to update their code to properly fix the bug of the instruction being played too late. It can still happen if `MapboxVoiceInstructionsPlayer` is created at the time when `play` is invoked.
b) what if it's not ready when the instruction is to be played? Should the user themselves right the code similar to the one I wrote?

But I agree that there is a possibility that it's not necessary to cancel queued announcements when the timer fires. Ideally, we should have some trigger saying that it doesn't make sense to play it now. But we don't have this now, or at least I'm not aware of it.

Reply:
> a) it would require users to update their code to properly fix the bug of the instruction being played too late. It can still happen if MapboxVoiceInstructionsPlayer is created when play is invoked.

I assume you are referring to the problem caused by `MapboxAudioGuidance.audioGuidanceFlow(...)` recreating `MapboxVoiceInstructionsPlayer` when the announcement language didn't match the language of the player (I believe you already patched that in your previous PRs).

That being said, since that `MapboxAudioGuidance` is at fault here, and this problem only occurs when `MapboxAudioGuidance` is being used, let's implement the fix that class instead of `MapboxVoiceInstructionsPlayer`. I don't think there will ever be a situation where a client has to cancel an individual instruction.

WDYT? @dzinad @abhishek1508

Reply:
Not exactly. If the user does not use Drop-In and has the following code:
```
    private val voiceInstructionsPlayer: MapboxVoiceInstructionsPlayer by lazy {
        MapboxVoiceInstructionsPlayer(...)
    }
    ...
    private val speechCallback =
        MapboxNavigationConsumer<Expected<SpeechError, SpeechValue>> { expected ->
            expected.fold(
                { error ->
                    voiceInstructionsPlayer.play(...)
                },
                { value ->
                    voiceInstructionsPlayer.play(...)
                }
            )
        }    
```
and the player is not accessed before that, there will be the same problem.

Reply:
Ok, but that's an implementation error that the developer should fix, not the SDK. We should discourage lazy initialization of the MapboxVoiceInstructionsPlayer for that exact reason. Adding a note to the https://docs.mapbox.com/android/navigation/guides/ui-components/voice/ should suffice.

Reply:
Not only the docs.
https://github.com/mapbox/mapbox-navigation-android/pull/6537
https://github.com/mapbox/mapbox-navigation-android-examples/pull/162
`android-docs/pull/2202`